cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -363,17 +363,6 @@ private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
 
         LOGGER.infoCr(reconciliation, "Rebalance action from state [{}]", currentState);
 
-        if (Annotations.isReconciliationPausedWithAnnotation(kafkaRebalance)) {
-            // we need to do this check again because it was triggered by a watcher
-            KafkaRebalanceStatus status = new KafkaRebalanceStatus();
-
-            Set<Condition> unknownAndDeprecatedConditions = validate(reconciliation, kafkaRebalance);
-            unknownAndDeprecatedConditions.add(StatusUtils.getPausedCondition());
-            status.setConditions(new ArrayList<>(unknownAndDeprecatedConditions));
-
-            return updateStatus(reconciliation, kafkaRebalance, status, null).compose(i -> Future.succeededFuture());
-        }
-
```

Comment:
If this was kept (as mentioned on the comment below), would we still need the `onReconciliationPaused()` method? Is the purpose of `onReconciliationPaused()` to retain the ConfigMap of the previous state?

Reply:
I guess maybe we can get rid of the --> `onReconciliationPaused()` and handle the case for `ReconciliationPaused` in the `computeNextStatus` method itself

Reply:
case ReconciliationPaused:
                return requestRebalance(reconciliation, host, apiClient, kafkaRebalance, true, rebalanceOptionsBuilder);

Reply:
Something like this maybe. What do you think @kyguy?

Reply:
I guess we need to handle it because we don't want the state to be shown as `New` when `reconciliation is paused` (previous behaviour). It should be `ReconciliationPaused`

Reply:
Looks good to me!